azext_iot/tests/iothub/message_endpoint/conftest.py

```diff
@@ -0,0 +1,464 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from time import sleep
+import pytest
+from azext_iot.common.embedded_cli import EmbeddedCLI
+from azext_iot.tests.settings import DynamoSettings
+from azext_iot.tests.generators import generate_generic_id
+from typing import Optional, TypeVar
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+SubRequest = TypeVar('SubRequest')
+Mark = TypeVar('Mark')
+cli = EmbeddedCLI()
+REQUIRED_TEST_ENV_VARS = ["azext_iot_testrg"]
+settings = DynamoSettings(req_env_set=REQUIRED_TEST_ENV_VARS)
+RG = settings.env.azext_iot_testrg
+
+
+def generate_hub_id() -> str:
+    return f"test-hub-{generate_generic_id()}"[:40]
+
+
+def generate_hub_depenency_id() -> str:
+    return f"testhubdep{generate_generic_id()}"[:24]
```

Comment:
are we ok with this naming for everything related to the hub? It will help with searching and mass deleting if needed

Reply:
I'm fine tagging things but maybe we make it a tiny bit more readable?
```suggestion
    return f"test_hub_dep_{generate_generic_id()}"[:24]
```

or possibly even link it to the hub it's being generated for:
```suggestion
    return f"{hub_id}_dependency_{generate_generic_id()}"[:24]
```

Reply:
There are two issues with this
1. not every service accepts _ (looking at event hubs) and some don't even accept non alphanumeric characters (looking at storage) in the name
2. I am also trying to keep the length shorter cause some services have shorter name limits than the iot hub

Reply:
my solution here was tagging - but I sorta removed that for this test and was planning on figuring out a better way in a future pr

Reply:
offline- decided aziotclitest as a prefix is better